src/steamship/agents/agents.py

```diff
@@ -0,0 +1,187 @@
+import uuid
+from abc import ABC, abstractmethod
+from typing import Any, List, Optional
+
+from pydantic import BaseModel
+
+from steamship import Block, File, Steamship, Tag, Task
+from steamship.invocable import InvocableResponse, PackageService, post
+from steamship.utils.kv_store import KeyValueStore
+
+
+class Action(BaseModel):
```

Comment:
Is this the "Noun which represents the doing of a thing?"

E.g. --- All agent behavior is composed as [a1, a2... an] in the presence of some AgentContext?

The ReAct reasoner, or python code, exists to enqueue future Actions. And future Actions only take the output of prior Actions as input?